src/test/java/com/teragrep/cfe18/controllerTests/CaptureMetaControllerTest.java

```diff
@@ -0,0 +1,335 @@
+        captureRelp.setProtocol("prot");
+        captureRelp.setFlow("capFlow");
+
+        String jsonFile = gson.toJson(captureRelp);
+
+        // forms the json to requestEntity
+        StringEntity requestEntity3 = new StringEntity(
+                String.valueOf(jsonFile),
+                ContentType.APPLICATION_JSON);
+
+        // Creates the request
+        HttpPut request3 = new HttpPut("http://localhost:" + port + "/capture/relp");
+        // set requestEntity to the put request
+        request3.setEntity(requestEntity3);
+        // Header
+        request3.setHeader("Authorization", "Bearer " + token);
+
+        HttpClientBuilder.create().build().execute(request3);
+
+        // ******************************************************************************************************
+
+        CaptureMeta captureMeta = new CaptureMeta();
+        captureMeta.setCapture_id(1);
+        captureMeta.setCapture_meta_key("relpKey1");
+        captureMeta.setCapture_meta_value("relpValue1");
+
+        String jsonFileApplication = gson.toJson(captureMeta);
+
+        // forms the json to requestEntity
+        StringEntity requestEntity4 = new StringEntity(
+                String.valueOf(jsonFileApplication),
+                ContentType.APPLICATION_JSON);
+
+        // Creates the request
+        HttpPut request4 = new HttpPut("http://localhost:" + port + "/capture/meta/");
+        // set requestEntity to the put request
+        request4.setEntity(requestEntity4);
+        // Header
+        request4.setHeader("Authorization", "Bearer " + token);
+
+        // Get the response from endpoint
+        HttpResponse httpResponse = HttpClientBuilder.create().build().execute(request4);
+
+        // Get the entity from response
+        HttpEntity entity = httpResponse.getEntity();
+
+        // Entity response string
+        String responseString = EntityUtils.toString(entity);
+
+        // Parsin respponse as JSONObject
+        JSONObject responseAsJson = new JSONObject(responseString);
+        // Creating expected message as JSON Object from the data that was sent towards endpoint
+        String expected = "New capture meta created for = 1";
+
+        // Creating string from Json that was given as a response
+        String actual = responseAsJson.get("message").toString();
+
+        // Assertions
+        assertThat(
+                httpResponse.getStatusLine().getStatusCode(),
+                equalTo(HttpStatus.SC_CREATED));
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @Order(2)
+    public void testRetrieveCaptureMeta() throws Exception {
+        ArrayList<CaptureMeta> expected = new ArrayList<>();
+        CaptureMeta captureMeta = new CaptureMeta();
+        captureMeta.setCapture_id(1);
+        captureMeta.setCapture_meta_key("relpKey1");
+        captureMeta.setCapture_meta_value("relpValue1");
+        expected.add(captureMeta);
+
+        String json = gson.toJson(expected);
```

Comment:
Theres no real self-contained guarantee that captureMeta works correctly -> can cause expected to be `{}` -> the responseGet can also return invalid values but this test still passes. Either build json from well known string or also test the captureMeta contents.